src/biogeochem/CNCIsoFluxMod.F90

```diff
@@ -1276,32 +1266,22 @@ subroutine CIsoFlux3(num_soilc , filter_soilc, num_soilp  , filter_soilp, &
 
 
       ! calculate the column-level flux of deadstem and deadcrootc to cwdc as the result of fire mortality.
-      do pi = 1,max_patch_per_col
-         do fc = 1,num_soilc
-            cc = filter_soilc(fc)
-            if ( pi <=  col%npatches(cc) ) then
-               pp = col%patchi(cc) + pi - 1
-               if (patch%active(pp)) then
-                  do j = 1, nlevdecomp
-                     iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) = &
-                          iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) + &
-                          (iso_cnveg_cf%m_deadstemc_to_litter_fire_patch(pp) + &
-                          iso_cnveg_cf%m_livestemc_to_litter_fire_patch(pp)) * &
-                          patch%wtcol(pp) * stem_prof(pp,j)
-                     iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) = &
-                          iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) + &
-                          (iso_cnveg_cf%m_deadcrootc_to_litter_fire_patch(pp) + &
-                          iso_cnveg_cf%m_livecrootc_to_litter_fire_patch(pp)) * &
-                          patch%wtcol(pp) * croot_prof(pp,j)
-                  end do
-               end if
-            end if
+      do fp = 1,num_soilp
+         pp = filter_soilp(fp)
+         cc = patch%column(pp)
+         do j = 1, nlevdecomp
+            iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) = &
+                 iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) + &
+                 (iso_cnveg_cf%m_deadstemc_to_litter_fire_patch(pp) + &
+                 iso_cnveg_cf%m_livestemc_to_litter_fire_patch(pp)) * &
+                 patch%wtcol(pp) * stem_prof(pp,j)
+            iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) = &
+                 iso_cnveg_cf%fire_mortality_c_to_cwdc_col(cc,j) + &
+                 (iso_cnveg_cf%m_deadcrootc_to_litter_fire_patch(pp) + &
+                 iso_cnveg_cf%m_livecrootc_to_litter_fire_patch(pp)) * &
+                 patch%wtcol(pp) * croot_prof(pp,j)
          end do
-      end do
 
```

Comment:
i like how you removed the soil filter, as this can all be done inside one filter, and the loop doesn't have to be broken up.

Reply:
Yeah, it looks like you were able to simplify this so only the patch filter is used rather than both a column and patch filter. That is a good improvement.

Reply:
Resolving.